### PR TITLE
test, watch, deploy updates

### PR DESCRIPTION
**Pull request by jamessimone**
fixing issue in deploy.js for dependent classes being invalid (now shows all classes with compilation errors instead of crapping out on the TableDeclaration property being null)

some minor householding tweaks (remove skipped deployment components from showing in log)

added --tests opt to choose to deploy with tests or not (since there was some functionality already written out to use this feature, but there was no way to access it previously)

### CODE REVIEW

File: commands/deploy.js [hunk 1]
```diff
@@ -359,17 +360,15 @@ function logDetails(details, opts) {
       .value();
   }
 
-  if(details.runTestResult) {
+  if(details.runTestResult.any) {
```
Comment (jamessimone) [inline commands/deploy.js]: needs to be changed back to the above line - .any can fail on null

**Review by bruce-lindsay — APPROVED** [review]
Hi james. finally getting around to review this. looks good to me. i think the only thing i'd consider enhancing is placement of the error for 'class is invalid' and need recompilation business. it's likely to be missed as it doesn't trigger the test failures boolean.

**Comment by bruce-lindsay** [comment]
further -- i tried this version and it is working well for me!

**Comment by bruce-lindsay** [comment]
hmm actually the output is a little funky in deploy --tests when there are tests failing. :face_with_head_bandage: 

```
[dmc] component successes [25] ====>
[dmc] [update] package.xml
[err] test results ====>
[dmc] * tests run: 472
[dmc] * failures: 6
[dmc] * total time: 82.199s
[err] undefined
[err] undefined
[err] undefined
[err] undefined
[err] undefined
[err] undefined
[err] metatadata api deployment failed
[dmc] [NOT OK]
```

**Pull request merged**